### PR TITLE
Simplify guards in List.ascii_printable?/2

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

File: lib/elixir/lib/list.ex [hunk 1]
```diff
@@ -500,39 +500,9 @@ defmodule List do
...
-  end
-
-  defp ascii_printable_guarded?([?\e | rest], counter) do
-    ascii_printable_guarded?(rest, decrement(counter))
-  end
-
-  defp ascii_printable_guarded?([?\a | rest], counter) do
+       # when char in '\a\b\t\n\v\f\r' or char == ?\e or char in 32..126 (ASCII printables)
+       when is_integer(char) and
+              ((char >= 7 and char <= 13) or char == 27 or (char >= 32 and char <= 126)) do
```
Comment (whatyouhide) [inline lib/elixir/lib/list.ex]:
```suggestion
              ((char >= 7 and char <= 13) or char == ?\e or (char >= 32 and char <= 126)) do
```

File: lib/elixir/lib/list.ex [hunk 2]
```diff
@@ -500,39 +500,9 @@ defmodule List do
...
-  defp ascii_printable_guarded?([?\f | rest], counter) do
-    ascii_printable_guarded?(rest, decrement(counter))
-  end
-
-  defp ascii_printable_guarded?([?\e | rest], counter) do
-    ascii_printable_guarded?(rest, decrement(counter))
-  end
-
-  defp ascii_printable_guarded?([?\a | rest], counter) do
+       # when char in '\a\b\t\n\v\f\r' or char == ?\e or char in 32..126 (ASCII printables)
```
Comment (whatyouhide) [inline lib/elixir/lib/list.ex]: I would rephrase this comment to make it human readable instead of code.

```suggestion
       # 7..13 is the range '\a\b\t\n\v\f\r'. 32..126 are ASCII printables.
```

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
